examples/ex1.cpp

```diff
@@ -192,7 +192,11 @@ int main(int argc, char *argv[])
    //    domain integrator.
    BilinearForm a(&fespace);
    if (pa) { a.SetAssemblyLevel(AssemblyLevel::PARTIAL); }
-   if (fa) { a.SetAssemblyLevel(AssemblyLevel::FULL); }
+   if (fa)
+   {
+      a.SetAssemblyLevel(AssemblyLevel::FULL);
+      a.EnableSparseMatrixSorting( Device::Allows(~Backend::CPU_MASK) );
```

Comment:
Just my 2c: I will prefer if `Device::Allows(~Backend::CPU_MASK)` is the default value and this is not visible to users in Example 1.

Reply:
Maybe we can add a differently named `Device` member function that makes this check in a more user-readable way?

I was about to suggest `Device::IsEnabled` but I see that we have a function like that already, which checks `mode == ACCELERATED`, which I **think** (but I am not sure) should do the right thing?

So we could have (maybe with a comment for clarity)
```c++
// Sort the matrix when running on GPU
a.EnableSparseMatrixSorting(Device::IsEnabled());
```

Reply:
I like that comment

Reply:
To be more precise, `Device::IsEnabled()` (which does seem to be equivalent to `Device::Allows(~Backend::CPU_MASK)`) is true for the OpenMP backends too, so the comment can be _"... when running on GPU or OpenMP"_ or something like that. Technically, the `debug` device is also included in this category but I'm not sure it's worth mentioning that.